plugin/NERD_commenter.vim

```diff
@@ -237,6 +237,7 @@ let s:delimiterMap = {
     \ 'java': { 'left': '//', 'leftAlt': '/*', 'rightAlt': '*/' },
     \ 'javacc': { 'left': '//', 'leftAlt': '/*', 'rightAlt': '*/' },
     \ 'javascript': { 'left': '//', 'leftAlt': '/*', 'rightAlt': '*/' },
+    \ 'javascript.jsx': { 'left': '//', 'right': '', 'leftAlt': '{/*', 'rightAlt': '*/}' },
```

Comment:
Is this a standard vim `filetype` code?

Reply:
@alerque Thanks for pointing that out (didn't check on that). Apparently this is non standard, however I've seen many plugins requiring setting these in order to function (other editors like vscode seem to use it too). I can totally understand if that is an issue though.

Reply:
Links? A few other major plugins that require this might be precedent enough. If we start adding non-vim standardized types to this list though there are a couple things I'd want to see changed. First I think I'd keep them in a separate array in the code or somehow clearly partitioned off so that it was clear what was vanilla filetypes and what is hacks. Secondly I think there should be some documentation explaining that some of the "supported" types need to be manually set as they are not things vim will set. Somewhere in the plugin setup should tip people off to not all file types being created equal.

Lastly if there are any discussions about _adding_ these file types to vim/nvim that would be immanently relevant too.

Reply:
I assume all plugins trying to fully support Microsofts language server protocol are affected by this. As you can see in the [LSP specification](https://microsoft.github.io/language-server-protocol/specification) there are filetypes named `javascriptreact` and `typescriptreact`. This is also the case for [VSCode](https://code.visualstudio.com/docs/languages/identifiers) since it is built upon that. 

Taking a look at my autocomplete plugin of choice (coc.nvim), they also seem to require to set either of those filetypes f.ex [(javascriptreact or javascript.jsx)](https://github.com/neoclide/coc.nvim/blob/b9967440b806a20067342b2ecc09682e6e1c685c/src/model/document.ts#L86) in order to have support for jsx/tsx specific snippets. This is also confirmed by a maintainer [here](https://github.com/neoclide/coc.nvim/issues/459#issuecomment-465658887).

Couldn't find an issue in the vim/nvim repos discussing this, so I might just create an issue over there. This would be the best way to avoid anything hacky / unofficial.